Suggested charter edits and additions (#255)
todogroup/governance: CHARTER.adoc

= Charter of The TODO Group

:toc:

 . *Mission of the TODO Group (“TODO”).*

>  The TODO Group’s mission is to identify key policy and process choices related to corporate open source engagement and create tools and educational materials that promote best practices around such engagements.

Reviewer: ```suggestion
 The TODO Group’s mission is to provide a community of practice and a structured way to identify key policies and processes related to organization's open source implementation and create tools and educational resources to aid them across its journey.
```

Reviewer 2: > 

The TODO Group's mission is to convene a community of practice focused on best practices for corporate open source programs. The Community of Practice will identify, document, and share policy and process best practices and create tools and educational materials. The group serves business professionals in OSPOs looking to collaborate with fellow TODO members and other related affinity groups and communities, including but not limited to NGOs, NPOs, educational institutions, or governments.

Reviewer 2: I do not agree. I am explicitly stating that TODO serves corporate members that collaborate with all these other groups.

Reviewer: final resolution from the SC members (5/7) during SC Charter call: update scope using @lhawthorn last definition:

*The TODO Group's mission is to convene a community of practice focused on best practices for corporate open source programs. The Community of Practice will identify, document, and share policy and process best practices and create tools and educational materials. The group serves business professionals in OSPOs looking to collaborate with fellow TODO members and other related affinity groups and communities, including but not limited to NGOs, NPOs, educational institutions, or governments.*

Reviewer 3: My thoughts here as we discussed the topic of TODO Group and having OSPOs outside the corporate world join the organization a few years ago when we posted this blog + FAQ:

https://todogroup.org/blog/becoming-a-todo-group-member-as-an-associate/
https://todogroup.org/faq/#:~:text=about%20membership%20benefits.-,As%20a%20university%20or%20academic%20institution%2C%20can%20I%20join%20the,discussions%20with%20experienced%20OSPO%20professionals.

The charter should reflect that we FOCUS on corporate OSPOs but don't exclude folks on joining the organization from outside the corporate world, we already have governments and academic institutions involved as official members.

Reviewer: What about something like:

*The TODO Group's mission is to convene a community of practice focused on best practices for open source programs. While our primary focus is on corporate OSPOs, we welcome members from outside the corporate world, including governments, NGOs, NPOs, educational institutions, and others. The Community of Practice will identify, document, and share policy and process best practices and create tools and educational materials. The general members serve professionals in OSPOs looking to collaborate with fellow TODO general members and other related affinity groups and communities. Our general members include a diverse group of organizations committed to advancing open source best practices and creating a more inclusive and collaborative open source ecosystem.*

Reviewer: From the last SC meeting: 

*leave the statement as it was initially. Update the section about the general membership*

>  . *Membership*

Reviewer: ```suggestion
 . *General Membership*
```

> The TODO Group membership is composed of individuals deeply involved in running open source programs setup to drive open source across divisions and product lines within their organization. These individuals have learned from practical experience what it takes to successfully engage in or create open source projects and build effective internal open source programs that guide how developers in their company engage in external communities. +

Reviewer 4: nit-pick

```suggestion
The TODO Group membership is composed of individuals deeply involved in running open source programs, tasked to drive open source across divisions and product lines within their organization. These individuals have learned from practical experience what it takes to successfully engage in or create open source projects and build effective internal open source programs that guide how developers in their company engage in external communities. +
```

Reviewer 2: LGTM

> The TODO Group shall be composed of the General Members and a Steering Committee. Anyone may participate as a General Member as long as they meet the General Member Qualifications. At such time as a General Member no longer meets the qualifications, their General Member status will be rescinded.

Reviewer: ```suggestion
The TODO Group's general membership shall be composed of the General Members and a Steering Committee. Anyone may participate as a General Member as long as they meet the General Member Qualifications. At such time as a General Member no longer meets the qualifications, their General Member status will be rescinded.
```

 .. General Member Qualifications: An individual may be qualified to join TODO Group as a General Member if they:
 ... Are an employee of an organization that is a member of The Linux Foundation at any membership level.
 ... Are an employee of an organization with development teams that use or contribute to open source projects across multiple divisions, products, or departments,
 ... in a role that manages, oversees, or guides the organization’s policies and contributions to open source projects (an “Open Source Program Office”),

>  ... and the organization does not derive a significant portion of its revenue as a solution provider to other OSPOs. +

Reviewer: on line 11: remove *TODO Group* ( they are free to participate in other community meetings in OSPOlogy, for instance). 

on line 12, change *company* for *organization*

Reviewer 2: I am confused by your first point. There is no TODO Group mentioned in line 11.

Reviewer 2: Perhaps add ..

... Are actively engaged as a project member contributing to and improving TODO's offerings, such as [perhaps noting scribe duties, hosting meetings, hosting TODO Group meetups]

Reviewer 2: It is clear from the website (but not charter) that individual's can join the community as long as they abide by the code of conduct. https://todogroup.org/associates/

Reviewer: @lhawthorn The charter only mentions general members and the steering committee. I suggest including a note at the beginning of the charter mentioning that this charter only covers general members and steering committee rights, criteria, etc, but the TODO Group structure covers more actors that are not included on the charter (OSPO associate organizations, individual contributors and OSPO practitioners, General member organizations and steering committee individuals)

>  ... If any committee member should resign or otherwise be unable to complete their term for any reason, with more than six (6) months remaining in their term, the remaining committee members shall appoint a replacement member. For election or appointment purposes, that member's term will end when the outgoing member's term would have ended.

Reviewer 5: ```suggestion
 ... If any committee member should resign or otherwise be unable to complete their term for any reason, with more than six (6) months remaining in their term, the remaining committee members may appoint a replacement member. For election or appointment purposes, that member's term will end when the outgoing member's term would have ended.
```

 ... If any committee member should resign or otherwise be unable to complete their term for any reason, with six (6) months or fewer remaining in their term, their position will be left unfilled until the regularly scheduled election or appointment process for that position.
 ... Elections shall be run using the Condorcet-IRV method through the Cornell online service (+++<u>+++http://civs.cs.cornell.edu/+++</u>+++). The methodology or service used in future elections may be changed via a five (5) member approval vote of current members of the committee at that time.
 .. Removal:
 ... Steering Committee members shall qualify for removal for:
 .... Violation of the TODO Group Code of Conduct.
 .... Inexcused absence from two (2) consecutive Steering Committee meetings.

>  .... Regular non-participation in Steering Committee votes and activities via email, real-time chat, or in person.

Reviewer: Maybe we can be more specific about what "regular" means here? for instance, *being inactive longer than three months* (no email responses, rea-time chat participation, monthly calls, etc)

Author: 100% agree! We definitely need to define that.

I advise that the "regular non-participation" limit be shorter than 3 months. That's 25% of a year when someone may have ghosted w/o the SC having recourse for taking action. 6-8 weeks would be my preference.

My expectation here is that along with these charter changes there'd be policies defined for how to go about some of these things. Policies/procedures are easier to evolve than the charter, after all. A policy/procedure for this non-participation thing could be something like "step 1: reach out the the SC member to see what's up; step 2: if no reply in $timeframe, SC shall draft and vote on a proposal to remove the member from the SC", etc.

Reviewer 4: I agree. How about 8 weeks, as this is somewhat aligned with missing 2 SC meetings?

```suggestion
 .... Non-participation in Steering Committee votes and activities via email, real-time chat, or in person for more than consecutive 8 weeks.
```

Reviewer 2: I support the eight week non-participation change, but would like us to consider whether we change the charter to allow for a new nominee to come from the member organization for the non-participating SC member.

In other words, if I am inactive for 8 weeks for e.g. a medical issue, but Red Hat would like to appoint someone to cover meetings in my stead, I would like to make this possible. I believe the replacement should be subject to SC approval. 

If no replacement is recommended, we would then move to working with the current charter guidelines - replacement by appointment of SC in first six months of term; no replacement required in last six months of term.

 ... Meetings will occur at least once per quarter.
 ... Details of meeting mechanisms and instructions for attendance will be published reasonably in advance to all members in good standing.
 .. Communication mechanisms

>  ... All public communication not during meetings will take place over the public group mailing list(s), real-time chat, and GitHub and will be accessible to members and non-members alike.

Reviewer 4: I suggest to use a vendor neutral term:

```suggestion
 ... All public communication not during meetings will take place over the public group mailing list(s), real-time chats, issue trackers, or source code management systems and will be accessible to members and non-members alike.
```

>  ... All public communication not during meetings will take place over the public group mailing list(s), real-time chats, issue trackers, or source code management systems and will be accessible to members and non-members alike.

Reviewer: ```suggestion
... Communication is divided between community (public) and general members (private). The general members at times share sensitive information and use a private channel for that type of discussion. Information shared in those channels is later anonymized and shared with the wider OSPO community via [OSPO Discussions](https://github.com/todogroup/ospology/discussions).

 ... All public communication not during meetings will take place over the public group mailing list(s), real-time chats, issue trackers, or source code management systems and will be accessible to members and non-members alike.
```

Reviewer 4: This is fine for me to include in the charter.

Reviewer 5: LGTM. Somewhat nit picky... Are all discussions from the private channel anonymized and shared in OSPO Discussions? If not, can say "Some of the information shared in those channels are later anonymized..."

Reviewer 2: LGTM with @annania's recommendation to say "some of the information" as it is an accurate synopsis of our operating processes. Please merge after that change has been made.

>  ... All private communication not during meetings will take place over the private group mailing list, real-time chat, and GitHub and access will be limited to members only.

Reviewer 4: I suggest to use vendor neutral terms:

```suggestion
 ... All private communication not during meetings will take place over the private group mailing list, real-time chats, issue trackers and source code management systems, and access will be limited to members only.
```

 .. Level of confidentiality
 ... All communication during meetings will be under+++<u>+++ Chatham House Rules+++</u>++++++<u>+++.+++</u>+++
 ... All private communication via all media as defined above will be restricted to members only and not permitted for public distribution.

>  ... All communication via all media as defined above will be public.

Reviewer 4: Nit-picking: "all communication" would include the private communcation as well. So, as a suggestion:

```suggestion
 ... All non-private communication via all media as defined above will be public.
```

Reviewer 6: @gkunz  The end of the sentence already defines communication will be public. Trying to understand the intent of this sentence - should it be 'All **other** communication via all media as defined above will be public.'?

Reviewer 6: Found it - in TODO original charter it was stated as "All communication on the public mailing list will be public"

Reviewer 4: @tsteenbe, yes, "all other" works well. I just wanted to add a clarifying word because the newly proposed "All communication via all media as defined above will be public." contradicts the statement about the private communication.

Reviewer 2: This does not cover how we use Slack to communicate nor does it clearly define the responsibilities of the SC for communicating about our work to the rest of the group (e.g. what is public, what is private). We should let people know there is private Slack channel for members, there is private Slack channel for the SC. There is nothing nefarious about it, so let's be transparent about how we have evolved to work.

Reviewer: @lhawthorn that is somehow documented in the [TODO structure mindmap](https://github.com/todogroup/governance/blob/main/TODO%20Structure/structure-mindmap.md#todo-groups-structure) and in the [community page](https://todogroup.org/community/). 

```
The general members at times share sensitive information and use a private mailing list for that type of discussion. Information shared in this channel is later anonymized and shared with the wider OSPO community via [OSPO Discussions](https://github.com/todogroup/ospology/discussions).

```

Does it make sense to be specific about the communication channels in the charter? Or should we be specific in other documentation (e.g community page) and keep the charter doc more general?

 . *Antitrust Guidelines*
 .. All members shall abide by The Linux Foundation Antitrust Policy available at: https://www.linuxfoundation.org/antitrust-policy/.
 .. All members may encourage open participation from any organization able to meet the membership requirements, regardless of competitive interests. Put another way, the TODO Group shall not seek to exclude members based on any criteria, requirements, or reasons other than those used for all members.
 . *General Rules and Operations*.

> The TODO Group shall be conducted so as to:

Reviewer: ```suggestion
The TODO Group General Members and broader community shall be conducted so as to:
```

Reviewer 2: I would like to best understand the exclusion criteria. I have seen organization's that appear to meet our criteria be told they cannot join TODO. Can we unpack this a bit in SC session?

 .. Under no circumstances shall The Linux Foundation be expected or required to undertake any action on behalf of TODO that is inconsistent with the tax exempt purpose of The Linux Foundation.
 . *Intellectual Property Policy*
 .. All contributions shall be accompanied by a Developer Certificate of Origin sign-off (http://developercertificate.org).
 .. All new inbound code contributions to TODO shall be made under the Apache License, Version 2.0 (available at http://www.apache.org/licenses/LICENSE-2.0) or an alternative OSI-approved open source license clearly identified by the contributor in the contribution in accordance with the DCO. All documentation will be contributed to and made available to TODO under the Creative Commons Attribution 4.0 International License (available at http://creativecommons.org/licenses/by/4.0/) or an alternative OSI-approved open source license clearly identified by the contributor in the contribution in accordance with the DCO.
 .. All outbound code will be made available under the license for the inbound contribution.

>  .. Any TODO member may raise an objection or concern regarding any contribution made under an alternative license to the Steering Committee for resolution.

Reviewer: ```suggestion
 .. Any TODO General Member may raise an objection or concern regarding any contribution made under an alternative license to the Steering Committee for resolution.
```

Reviewer 2: What is a contribution as far as a DCO is concerned? Should we specify code here? I have never executed a DCO and I am clearly contributing to TODO by adding to this repo.
